osu.Game.Rulesets.Catch/Difficulty/Skills/Movement.cs

```diff
@@ -57,49 +58,71 @@ protected override double StrainValueOf(DifficultyHitObject current)
                 catchCurrent.NormalizedPosition + (normalized_hitobject_radius - absolute_player_positioning_error)
             );
 
+            bool isDoubleHdash = catchCurrent.LastObject.HyperDash && catchCurrent.BaseObject.HyperDash;
+
             float distanceMoved = playerPosition - lastPlayerPosition.Value;
 
-            double weightedStrainTime = catchCurrent.StrainTime + 13 + (3 / catcherSpeedMultiplier);
+            double weightedStrainTime = catchCurrent.StrainTime / catcherSpeedMultiplier;
+            double edgeDashBonus = 0;
 
-            double distanceAddition = (Math.Pow(Math.Abs(distanceMoved), 1.3) / 510);
-            double sqrtStrain = Math.Sqrt(weightedStrainTime);
+            double distanceAddition;
 
-            double edgeDashBonus = 0;
+            // Counting direction changes for length scaling in the performance calculation
+            if (Math.Sign(distanceMoved) != Math.Sign(lastDistanceMoved) && Math.Sign(lastDistanceMoved) != 0 && Math.Abs(distanceMoved) > 4)
+            {
+                DirectionChangeCount += 1;
+            }
+
+            // Separating HyperDashes and basic fruits calculations
 
-            // Direction change bonus.
-            if (Math.Abs(distanceMoved) > 0.1)
+            // Basic fruit calculation
+            if (!catchCurrent.LastObject.HyperDash)
             {
-                if (Math.Abs(lastDistanceMoved) > 0.1 && Math.Sign(distanceMoved) != Math.Sign(lastDistanceMoved))
-                {
-                    double bonusFactor = Math.Min(50, Math.Abs(distanceMoved)) / 50;
-                    double antiflowFactor = Math.Max(Math.Min(70, Math.Abs(lastDistanceMoved)) / 70, 0.38);
+                // The base value is a ratio between distance moved and strain time
+                // For basic fruits, the distance is usually slightly more important for the calculation than the strain time
+                distanceAddition = 0.06 * Math.Pow(Math.Pow(Math.Abs(distanceMoved), 1) / weightedStrainTime, 2) / Math.Pow(Math.Max(1, catcherSpeedMultiplier), 2);
```

Comment:
`catchSpeedMultiplier` is used to nerf double time spacings. I didn't really want it to buff HT, but I guess it makes more sense like that so I will remove the cap.  And the calculation is something like `(Distance/Time/Speed)^2` so I will refactor it as `(D/(T*S))^2` so it's a bit more clear in my opition. I'm not sure what you want to know with a dimensional analysis, I don't really consider units, but you've got `distanceMoved` which is in pixels I guess (or some sort or normalized length), `weightedStrainTime` which I will change into `strainTime` because it's the same now, and that's in seconds. `catchSpeedMultiplier` is a factor, 0.5 for HT, 1.5 for DT and 1 otherwise. So basically I square the ratio of Distance/Time/Speed.